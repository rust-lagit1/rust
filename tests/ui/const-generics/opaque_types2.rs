#![feature(type_alias_impl_trait)]

type Foo = impl Sized;
//~^ ERROR: cycle detected when computing type of `Foo::{opaque#0}`

fn foo<const C: u32>() {}

const C: Foo = 42;

fn bar()
where
    Foo:,
{
    foo::<C>();
}

fn main() {}
