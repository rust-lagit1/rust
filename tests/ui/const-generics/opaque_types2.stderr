error[E0391]: cycle detected when computing type of `Foo::{opaque#0}`
  --> $DIR/opaque_types2.rs:3:12
   |
LL | type Foo = impl Sized;
   |            ^^^^^^^^^^
   |
note: ...which requires computing type of opaque `Foo::{opaque#0}`...
  --> $DIR/opaque_types2.rs:3:12
   |
LL | type Foo = impl Sized;
   |            ^^^^^^^^^^
note: ...which requires type-checking `bar`...
  --> $DIR/opaque_types2.rs:9:1
   |
LL | / fn bar()
LL | | where
LL | |     Foo:,
   | |_________^
note: ...which requires evaluating type-level constant...
  --> $DIR/opaque_types2.rs:7:1
   |
LL | const C: Foo = 42;
   | ^^^^^^^^^^^^
note: ...which requires const-evaluating + checking `C`...
  --> $DIR/opaque_types2.rs:7:1
   |
LL | const C: Foo = 42;
   | ^^^^^^^^^^^^
note: ...which requires caching mir of `C` for CTFE...
  --> $DIR/opaque_types2.rs:7:1
   |
LL | const C: Foo = 42;
   | ^^^^^^^^^^^^
note: ...which requires elaborating drops for `C`...
  --> $DIR/opaque_types2.rs:7:1
   |
LL | const C: Foo = 42;
   | ^^^^^^^^^^^^
   = note: ...which requires normalizing `Foo`...
   = note: ...which again requires computing type of `Foo::{opaque#0}`, completing the cycle
note: cycle used when checking that `Foo::{opaque#0}` is well-formed
  --> $DIR/opaque_types2.rs:3:12
   |
LL | type Foo = impl Sized;
   |            ^^^^^^^^^^
   = note: see https://rustc-dev-guide.rust-lang.org/overview.html#queries and https://rustc-dev-guide.rust-lang.org/query.html for more information

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0391`.
